src/sui-autocompleted/index.jsx

```diff
@@ -26,19 +26,55 @@ export default class Autocompleted extends Component {
     }
   }
 
+  isLastSelected = () => {
+    const { active } = this.state
+    const { suggests, withSections } = this.props
+    const lastIndex = suggests.length - 1
+    const lastSection = withSections ? lastIndex : 0
+    const lastPosition = withSections ? suggests[lastIndex].suggestions.length - 1 : lastIndex
+    return active.section === lastSection && active.suggestion === lastPosition
+  }
+
+  isFirstSelected = () => {
+    const { active } = this.state
+    return active.section === this.defaultPosition.section &&
+      active.suggestion === this.defaultPosition.suggestion
+  }
+
+  getLastSectionSuggestion = section => this.props.withSections ? this.props.suggests[section].suggestions.length - 1 : this.props.suggests.length - 1
+
   moveDown = () => {
     const { active } = this.state
-    const lastPosition = this.props.suggests.length - 1
-    return active === lastPosition
-      ? active
-      : active + DELTA_MOVE
+    const isLastSelected = this.isLastSelected()
+    if (isLastSelected) {
+      return active
+    }
+
+    return active.suggestion === this.getLastSectionSuggestion(active.section) ? {
+      section: active.section + DELTA_MOVE,
+      suggestion: 0
+    } : {
+      section: active.section,
+      suggestion: active.suggestion + DELTA_MOVE
+    }
   }
 
   moveUp = () => {
     const { active } = this.state
-    return active === this.defaultPosition
-      ? active
-      : active - DELTA_MOVE
+    const isFirstSelected = this.isFirstSelected()
+    if (isFirstSelected) {
```

Comment:
No need for an extra const since it's not used anywhere else, so you could just:
```js
if (this.isFirstSelected()) {
```